ui/components/schema/StructureView.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from "react";
+import {
+    Alert, ListView, ListViewItem, Grid, Row, Col
+} from "patternfly-react";
+import { EmptyStructureView } from "./EmptyStructureView";
+import style from "./structureView.css";
+
+// Graphql internal types that we don't want to render
+const wellKnownTypes = [
+    "String",
+    "Boolean",
+    "Int",
+    "__Schema",
+    "__Type",
+    "__TypeKind",
+    "__Field",
+    "__InputValue",
+    "__EnumValue",
+    "__Directive",
+    "__DirectiveLocation"
+];
+
+class StructureView extends Component {
+
+    renderEmpty() {
+        return <EmptyStructureView />;
+    }
+
+    renderAdditionalInfo(type) {
+        const { fields } = type;
+        if (!fields) {
+            return "n.a.";
+        }
+        return type.fields.length + (type.fields.length > 1 ? " fields" : " field");
+    }
+
+    renderFields(fields) {
+        if (!fields) {
+            // Some types won't have fields
+            return <div />;
+        }
+
+        return fields.map(field => {
+            const key = field.type.name || field.type.kind;
+            return (
+                <Row key={key + field.name} className={style["structure-item-row"]}>
+                    <Col xs={6} md={6}>
+                        {field.name}
+                    </Col>
+                    <Col xs={6} md={6}>
+                        {key}
+                    </Col>
+                </Row>
+            );
+        });
+    }
+
+    renderType(type, index) {
```

Comment:
IMO this is clear indication that we should extract another class from here. But it's all right at the moment, we can leave it like this.